clang/test/CXX/drs/cwg22xx.cpp

```diff
@@ -169,10 +169,35 @@ B b;
 // since-cxx11-error@-1 {{call to implicitly-deleted default constructor of 'B'}}
 //   since-cxx11-note@#cwg2273-B {{default constructor of 'B' is implicitly deleted because base class 'A' has a deleted default constructor}}
 //   since-cxx11-note@#cwg2273-A {{'A' has been explicitly marked deleted here}}
+
+struct X {
+  X(float); // #cwg2273-X
+  X(void*, int = 0) = delete;
+  X(int, const int*) = delete;
+  X(unsigned, int()) = delete;
+};
+
+struct Y : X {
+  using X::X; // #cwg2273-Y1
+  Y(double); // #cwg2273-Y2
+  Y(void* const, long = 1);
+  using IA = int[42];
+  Y(int, const IA);
+  Y(unsigned, int (*)(void));
```

Comment:
I'd appreciate other test cases of:
```
Y(int n, const int[n]);
Y(unsigned, const int (*)(void));
```
where the types are not the same. (The VLA test may have to live somewhere outside of the DR testing because that's an extension.) Speaking of extensions:
```
struct X {
  X(int) = delete;
};

struct Y : X {
  using X::X;
  Y(_Atomic int);
};

Y y = 1;
```
Those have different unqualified types because `_Atomic` is not stripped as a qualifier because it's Special™.